test/it/crdvalidation/etcdcopybackupstask/helper.go

```diff
@@ -0,0 +1,96 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package etcdcopybackupstask
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/gardener/etcd-druid/test/it/setup"
+	"github.com/gardener/etcd-druid/test/utils"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/gomega"
+)
+
+const testNamespacePrefix = "etcdcopybackupstask-validation-test"
+
+var (
+	itTestEnv setup.DruidTestEnvironment
+)
+
+func validatePatchedObjectCreation[T client.Object](g *WithT, patchedObject *unstructured.Unstructured, expectErr bool) {
+	cl := itTestEnv.GetClient()
+	ctx := context.Background()
+
+	if expectErr {
+		g.Expect(cl.Create(ctx, patchedObject)).NotTo(Succeed())
+		return
+	}
+	g.Expect(cl.Create(ctx, patchedObject)).To(Succeed())
+
+	// Ensure the object is created successfully.
+	// This needs to be done to ensure that 1) the object is created and 2) the created object is valid.
+	// If the duration validation is not correct, the object will be created but will fail to decode correctly.
+	// Reference: https://github.com/kubernetes/apimachinery/issues/131
+	g.Eventually(func() error {
+		fetchedObject := patchedObject.DeepCopyObject().(client.Object)
+		if err := cl.Get(ctx, client.ObjectKeyFromObject(patchedObject), fetchedObject); err != nil {
+			return fmt.Errorf("failed to fetch object: %w", err)
+		}
+		unstructuredObject, ok := fetchedObject.(*unstructured.Unstructured)
+		if !ok {
+			return fmt.Errorf("failed to convert fetched object to unstructured object: %T", fetchedObject)
+		}
+		typedObject := new(T)
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredObject.Object, typedObject); err != nil {
+			return fmt.Errorf("failed to convert unstructured object to %T: %w", typedObject, err)
+		}
+		return nil
+	}, "5s", "1s").Should(Succeed())
+}
+
+// Helper to patch a key within the given Kubernetes object. This is needed to simulate invalid values
+// passed in through YAML to the api-server.
+func patchObject[T client.Object](object T, path []string, value interface{}) *unstructured.Unstructured {
```

Comment:
Can you rename `path` to `fieldPath` so it is makes it easier to the reader?
```suggestion
// Helper to patch a key within the given Kubernetes object. This is needed to simulate invalid values
// passed in through YAML to the api-server.
func patchObject[T client.Object](object T, fieldPath []string, value interface{}) *unstructured.Unstructured {
```